design/ironic-interaction/ironic-background-docs.md

```diff
@@ -0,0 +1,346 @@
+   which contains configuration metadata and user defined "user-data".
+
+   1) Create a folder called “TEMPDIR”
+   2) In the case of ignition based configuration, that file would be
+      placed renamed "user_data" and placed in TMPDIR/openstack/latest/
+      folder.
+   3) Metadata for networking configuration setup using "cloud-init" or
+      cloud-init similar application would also be written to the
+      TEMPDIR/openstack/latest as well. This is out of scope, but is well
+      documented in the OpenStack community.
+   4) TEMPDIR is converted to an iso9660 image using a label of “config-2”.
+   5) The resulting ISO9660 image file is then compressed using the gzip
+      algorithm.
+   6) The resulting gzip image file is then encoded in base64 for storage
+      and transport. Ironic does the needful to decode and uncompress the
+      configuration drive prior to deployment.
+
+
+4. Send a HTTP POST to /v1/nodes/node-name/states/provision to initiate
+   the deployment
+
+       {“target”: “active”,
+        “configdrive”: “http://url-to-config-drive/node.iso.gz”}
+
+   Once the request to make the node active has been received by ironic,
+   it will proceed with the deployment process and execute the required
+   steps to help ensure the baremetal node reboots into the requested
+   disk image.
+
+### How to unprovision a baremetal node?
+
+A provisioned, or "active" baremetal node can be unprovisioned by sending
+a state change request to the ironic api. This request will move the
+baremetal node through the "cleaning" process which ironic utilizes to
+erase the contents of the disks. This can be a time in tensive process,
+and ultimately may only be useful for cleaning metadata except in limited
+circumstances.
+
+    PUT /v1/nodes/node-name/states/provision
+    { "target": "deleted" }
+
+### How to delete a baremetal node?
+
+Deletion of a node in ironic is removal from its inventory.
+
+As ironic is designed to manage the lifecycle of baremetal nodes, protections
+exist to prevent users of the ironic API from deleting nodes in states that
+may not be ideal. Mainly this restricts deletion to states where the node
+is not in use or is not actively performing a task.
+
+   Safe states are:
+   * "available"
+   * "manageable"
+   * "enroll"
+   * "adopt fail"
+
+This may be overridden by putting the node into "maintenance", during
+which ironic will not attempt to perform any operations.
+
+    PUT /v1/nodes/node-name/maintenance
+    DELETE /v1/nodes/node-name
+
+**NOTE:** Care should be taken to avoid triggering the deletion of a node in
+a "clean*" state.
+
+Additional information on states and state transitions in ironic can
+be found at: https://docs.openstack.org/ironic/latest/contributor/states.html
+
+### How to create the record of an active node
+
+Ironic possesses the functionality to create a node and move it into
+an "active" state. This is useful to create the record of an "active" node
```

Comment:
I would add: 'directly from "manageable" state' or rephrase with something like that.